src/rapidhtml/app.py

```diff
@@ -20,17 +38,43 @@ class RapidHTML(Starlette):
         - HTMX
     """
 
-    def __init__(self, *args, html_head: typing.Iterable = None, **kwargs) -> None:
+    """
+    Initializes the RapidHTML application.
+
+        Args:
+            html_head (typing.Iterable, optional): Tags to inject into each 
+                page's <head>. Defaults to None.
+            reload (bool, optional): Enables live-reloading. Defaults to False.
+    """
+
+    def __init__(
+        self, *args, html_head: typing.Iterable = None, reload: bool = False, **kwargs
+    ) -> None:
         super().__init__(*args, **kwargs)
-        if not html_head:
-            html_head = ()
+
+        self.reload = reload
         self.html_head = (Script(src="https://unpkg.com/htmx.org@2.0.1"),) + tuple(
-            html_head
+            html_head or ()
         )
-        self.router = RapidHTMLRouter(html_head=self.html_head)
 
-    def serve(self, *args, **kwargs):
-        uvicorn.run(self, *args, **kwargs)
+        if reload:
+            self.html_head += (Script(JS_RELOAD_SCRIPT),)
+            self.router = RapidHTMLRouter(html_head=self.html_head)
+            self.router.add_websocket_route("/live-reload", _ReloadSocket)
+        else:
+            self.router = RapidHTMLRouter(html_head=self.html_head)
+
+    def serve(self, appname=None, host="127.0.0.1", port=8001, *args, **kwargs):
```

Comment:
I think it should be 8000. It's the default for uvicorn and I initially used 8001 due to the port being taken and messed up by pushing the wrong example code
```suggestion
    def serve(self, appname=None, host="127.0.0.1", port=8000, *args, **kwargs):
```